### PR TITLE
Fix issue #1292

### LINKED ISSUE #1292: build error when compiling with clang++ and GMP
There is a problem when compiling examples/arithmetic/approximation.cpp and convergents.cpp.
Seems related to the fact that template static data member must be defined now in headers.

> In file included from /home/travis/build/DGtal-team/DGtal/examples/arithmetic/approximation.cpp:54:
In file included from /home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.h:600:
/home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.ih:739:8: error: instantiation of variable 'DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::singleton' required here, but no definition is available [-Werror,-Wundefined-var-template]
  if ( singleton == 0 )
       ^
/home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.ih:551:29: note: in instantiation of member function 'DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::instance' requested here
  else if ( this->myNode == instance().myOneOverZero )
                            ^
/home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.ih:521:3: note: in instantiation of member function 'DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::Fraction::pushBack' requested here
  pushBack( quotient );
  ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_iterator.h:445:13: note: in instantiation of member function 'DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::Fraction::push_back' requested here
        container->push_back(std::move(__value));
                   ^
/home/travis/build/DGtal-team/DGtal/examples/arithmetic/approximation.cpp:101:17: note: in instantiation of member function 'std::back_insert_iterator<DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::Fraction>::operator=' requested here
      *itback++ = std::make_pair( u, i++ );
                ^
/home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.h:548:32: note: forward declaration of template entity is here
    static LighterSternBrocot* singleton;
                               ^
/home/travis/build/DGtal-team/DGtal/src/DGtal/arithmetic/LighterSternBrocot.ih:739:8: note: add an explicit instantiation declaration to suppress this warning if 'DGtal::LighterSternBrocot<long, long, DGtal::StdMapRebinder>::singleton' is explicitly instantiated in another translation unit
  if ( singleton == 0 )
       ^
1 error generated.
make[2]: *** [examples/arithmetic/CMakeFiles/approximation.dir/approximation.cpp.o] Error 1
make[1]: *** [examples/arithmetic/CMakeFiles/approximation.dir/all] Error 2

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
*Thanks a lot for contributing to DGtal, before submitting your PR, please fill up the description and make sure that all checkboxes are checked. Please remove these lines in your PR.*

# PR Description

Fix  #1292 , build error with clang++ and GMP in package arithmetic.

# Checklist

- [NA] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [NA] Doxygen documentation of the code completed (classes, methods, types, members...)
- [NA] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/arithmetic/ModuleSRC.txt [hunk 1]
```diff
@@ -1,8 +1,4 @@
 ## Sources associated to the module math
 ##
 
-SET(DGTAL_SRC ${DGTAL_SRC} 
-		DGtal/arithmetic/ModuloComputer
-		DGtal/arithmetic/SternBrocot
-		DGtal/arithmetic/LightSternBrocot
-		DGtal/arithmetic/LighterSternBrocot)
+## SET(DGTAL_SRC ${DGTAL_SRC} ... )
```
Comment (dcoeurjo) [inline src/DGtal/arithmetic/ModuleSRC.txt]: maybe we could just remove this file..

**Comment by JacquesOlivierLachaud** [comment]
Ok. Done.

**Comment by dcoeurjo** [comment]
thx.. waiting for the bots and I merge.

**Comment by dcoeurjo** [comment]
Thanks merging

**Pull request merged**